lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2088,7 +2088,11 @@ void IRGenDebugInfoImpl::emitImport(ImportDecl *D) {
     return;
 
   assert(D->getModule() && "compiler-synthesized ImportDecl is incomplete");
-  ModuleDecl::ImportedModule Imported = {D->getModulePath(), D->getModule()};
+  // FIXME: Refactoring revealed that the next line was using
+  // ModuleDecl::ImportedModule inconsistently with typical use (placing the
+  // module path in the `accessPath` field). It's not immediately clear if there
+  // are matching inconsistencies elsewhere in IRGenDebugInfo.
+  ModuleDecl::ImportedModule Imported = { ImportPath::Access(), D->getModule() };
```

Comment:
I don't think anything here reads the `accessPath` field (so why are we passing `ImportedModule`s around instead of just `ModuleDecl*`s?), but I could use an extra eye on this.

Reply:
@adrian-prantl Can you comment on this?

Reply:
This is used only in IRGenDebugInfoImpl::finalize() to avoid creating a debug info breadcrumb for a module import twice. If you are saying this use is inconsistent, I'd image that it likely doesn't work correctly at the moment. We probably didn't notice, since a duplicate import in the debug info won't cause any problems other than wasting space.

Can you remind me what the access path is in a typical module import example? This was never really clear to me since Swift uses a flat namespace for modules. Or is that changing now?

Reply:
In a scoped import (like `import class Foo.Bar`), the access path is the declaration the import is scoped to (`Bar`). Most imports are not scoped (like `import Foo`); their access paths are empty.

If you’re trying to record what was imported so that you can fully(ish) recreate it later, you probably want to write `D->getAccessPath()` instead of `ImportPath::Access()` here. If you just want to record the modules and don’t care about the access paths, what’s here is fine.

Reply:
Okay, that makes sense! At the moment, we only record imports on a module granularity in the debug info <rdar://problem/68149326>, so I think having an empty access path is actually what we'd want here!

Reply:
Since you might eventually decide to start recording access paths, I think I'll just grab the right value from the ImportDecl and pass it along. Currently you'll just ignore it, but if that changes in the future you won't need to update this code.

Reply:
I've force-pushed a change to construct it with the correct (ignored) access path.